opt_einsum/parser.py

```diff
@@ -262,10 +262,16 @@ def convert_interleaved_input(operands: List[Any]) -> Tuple[str, List[Any]]:
     return subscripts, operands
 
 
-def parse_einsum_input(operands: Any) -> Tuple[str, str, List[ArrayType]]:
+def parse_einsum_input(*operands: Any, shapes: bool = False) -> Tuple[str, str, List[ArrayType]]:
```

Comment:
```suggestion
def parse_einsum_input(operands: Any, shapes: bool = False) -> Tuple[str, str, List[ArrayType]]:
```

We shouldn't have to change the operands unpacking here. The logs are pretty long, but I suspect this is the key issue.

Reply:
Yea, I think my latest change should fix this